src/Symfony/Component/EventDispatcher/InvokableEventSubscriberTrait.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait InvokableEventSubscriberTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        $method = new \ReflectionMethod(self::class, '__invoke');
+
+        if (0 === $method->getNumberOfParameters()) {
+            throw new \LogicException(sprintf('The Event name cannot be inferred from "%s::%s()" method without parameters, you must define one and type-hint the event class.', self::class, $method->getName()));
+        }
+
+        $parameter = $method->getParameters()[0];
+
+        if (null === $type = $parameter->getType()) {
+            throw new \LogicException(sprintf('The Event name cannot be inferred from "%s::%s()" method without type-hint the "$%s" parameter, you must type-hint the event class.', self::class, $method->getName(), $parameter->getName()));
+        }
+
+        if ($type->isBuiltin()) {
+            throw new \LogicException(sprintf('The "$%s" parameter defined in "%s::%s()" method must be a class, "%s" given.', $parameter->getName(), self::class, $method->getName(), $type->getName()));
+        }
+
+        return [$type->getName() => '__invoke'];
```

Comment:
I'd suggest writing this as `eturn [$type->getName() => ['__invoke', 0]];` just to have a reminder of the syntax when the priority is needed.

Reply:
He just changed it after I suggested the shorter syntax. 🙈 😓

Reply:
I saw that, this piece of code is far from trivial :)